From 913499026bb2d167eba32641df15429da4f5118c Mon Sep 17 00:00:00 2001
From: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
Date: Tue, 21 Nov 2023 14:29:12 +0530
Subject: [PATCH] wifi: ath12k: start monitor iface when ap + mon coexist

When monitor iface is created followed by an ap iface, mac80211 will
delete the existing ap iface to change the encap/decap to nwifi. While
deleting the ap iface MONITOR_CONF_ENABLED is cleared in
ath12k_mac_remove_link_interface. This is causing monitor mode to not
to create and start.

Fix by clearing the MONITOR_CONF_ENABLED flag only through ath12k_mac_op_config.

Signed-off-by: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 57528c089cf0..b2e2cf765c94 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -11212,7 +11213,6 @@ err_vdev_del:
 	/* Recalc txpower for remaining vdev */
 	ath12k_mac_txpower_recalc(ar);
 	ath12k_mac_ap_ps_recalc(ar);
-	clear_bit(MONITOR_CONF_ENABLED, &ar->monitor_flags);
 
 	ath12k_debugfs_remove_interface(arvif);
 
-- 
2.38.0

